### PR TITLE
Docs/x road

### PR DESCRIPTION
**Pull request by bthj**
# Description

Adding X-Road documentation pertaining to the Digital Iceland instance Straumurinn.

# Notes

This is a manual transfer of documentation initially set up in another GitBook space.  GitBook assets were added to `.gitbook/assets/x-road` in this project.  Is that suitable?

### CODE REVIEW

File: handbook/technical-overview/x-road/ArchitectureGuidelinesforServiceProvidersandConsumers.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+### Further reading <a id="d33a23fc-c9ce-497f-9957-61de4700f10a"></a>
+
+* [X-Road Architecture](https://x-road.global/architecture)
+* [X-Road Security Architecture](https://github.com/nordic-institute/X-Road/blob/develop/doc/Architecture/arc-sec_x_road_security_architecture.md)
+* [X-Road Architecture - Technical Specification](https://github.com/nordic-institute/X-Road/blob/develop/doc/Architecture/arc-g_x-road_arhitecture.md)
+
+## Availability <a id="2c8b484c-479c-4a28-9dc0-211bc028a9c9"></a>
+
+A benefit of the distributed design of X-Road is that no one component is a system-wide bottleneck or point of failure. The availability of security services for each service consumer and provider can be increased with redundant configurations. There are two possible approaches to increase the availability of X-Road security servers: Internal and external load balancing. Those two approaches entail a trade-off between simplicity and flexibility.
```
Comment (darrikonn) [inline handbook/technical-overview/x-road/ArchitectureGuidelinesforServiceProvidersandConsumers.md:23]: `no one component`?

File: handbook/technical-overview/x-road/SecurityServerInstallationAndRegistrationSteps.md [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+* Add OS user to run the X-Road server: `sudo useradd --system --home /var/lib/xroad --no-create-home --shell /bin/bash --user-group --comment "X-Road system user" xroad`  If that user will be used for interactive SSH log-ins, then so the Security Server PIN won't be cleared \(even though auto-login is configured\), the following command can be entered: `loginctl enable-linger xroad`
+* During installation, dialog will appear asking for host and IP information for certificate generation. The latter set of the dialog will be for configuring certificates for the xroad-proxy-ui-api. Here it may be desirable to change the value from the auto-detected machine host name to a domain name used for accessing the Admin UI:
+
+  ![A screenshot of a cell phone
+
+  Description automatically generated](../../../.gitbook/assets/x-road/0%20%282%29.png)
+
+  ![A screenshot of a cell phone
+
+  Description automatically generated](../../../.gitbook/assets/x-road/1%20%281%29.png)
```
Comment (darrikonn) [inline handbook/technical-overview/x-road/SecurityServerInstallationAndRegistrationSteps.md:38]: These won't render correctly (newline is breaking it)

File: handbook/technical-overview/x-road/SecurityServerInstallationAndRegistrationSteps.md [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+
+* During a [Security Server initial configuration](https://github.com/nordic-institute/X-Road/blob/develop/doc/Manuals/ig-ss_x-road_v6_security_server_installation_guide.md#3-security-server-initial-configuration), a PIN is required to be set, which should be a 12 digit, alpha-numeric password: [https://en.wikipedia.org/wiki/Personal\_identification\_number](https://en.wikipedia.org/wiki/Personal_identification_number) [https://en.wikipedia.org/wiki/ISO\_9564\#PIN\_length](https://en.wikipedia.org/wiki/ISO_9564#PIN_length)
+* Disable message payload logging \(for now, due to GDPR\).
+
+  The xroad-securityserver-is variant has the message logging disabled by default, from X-Road version 6.24.0 onwards.  
+
+* For PIN to be entered automatically when starting X-Road services, follow this guide: [https://github.com/nordic-institute/X-Road/blob/develop/doc/Manuals/Utils/ug-autologin\_x-road\_v6\_autologin\_user\_guide.md](https://github.com/nordic-institute/X-Road/blob/develop/doc/Manuals/Utils/ug-autologin_x-road_v6_autologin_user_guide.md)
+* Starting and stopping X-Road services to test the PIN entry functionality:
+
+```python
```
Comment (darrikonn) [inline handbook/technical-overview/x-road/SecurityServerInstallationAndRegistrationSteps.md:63]: This is `bash` :)

**Review by darrikonn — CHANGES_REQUESTED** [review]
Nice!

Minor comments

File: handbook/technical-overview/x-road/SecurityServerInstallationAndRegistrationSteps.md [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+curl -H "X-Road-Client: IS-DEV/GOV/10000/island-is-client" "http://localhost:8080/r1/IS-DEV/GOV/10006/Skatturinn-Protected/APIS-v1/company?name=skatturinn"
+```
+
+IS-**TEST**
+
+**Ísland.is to Skatturinn:**
+
+```python
```
Comment (jeremybarbet) [inline handbook/technical-overview/x-road/SecurityServerInstallationAndRegistrationSteps.md:262]: `bash` instead of python in this file

**Review by jeremybarbet — CHANGES_REQUESTED** [review]
- Some codeblocks are using `python` syntax highlight instead of `bash`

- I opened a [PR](https://github.com/island-is/island.is/pull/2443) to move from `.gitbook/assets` to folder-assets. I believe we gonna end up with a huge `.gitbook/assets` folder at some point and it might be better organised if we have each assets by folder. Also, the folder is supposedly generated when editing the content through GitBook directly but we are only editing file through git, so it might be confusing. I'm not against changing back my PR if you believe it's easier to have the `.gitbook/assets`, it was just my reasoning behind this change

File: handbook/technical-overview/x-road/SecurityServerInstallationAndRegistrationSteps.md (hunk 4 above)
Comment (bthj) [reply handbook/technical-overview/x-road/SecurityServerInstallationAndRegistrationSteps.md]: Sounds good to keep the assets close to the documentation they relate to.

**Comment by jeremybarbet** [comment]
LGTM. Looks like the SUMMARY.md file is outdated, you only need to run `yarn gitbook` and commit the file changes

**Review by jeremybarbet — DISMISSED** [review]
LGTM

**Comment by bthj** [comment]
@darrikonn , would you consider the requested changes as implemented?

**Comment by darrikonn** [comment]
Yuup looks good!. You can ping me when the build is green, and then I'll approve it 👍

**Comment by bthj** [comment]
> Yuup looks good!. You can ping me when the build is green, and then I'll approve it 👍

Hjælp!  It's not clear to me what needs to be done to green-light this :o)

**Comment by darrikonn** [comment]
Just need to format `handbook/technical-overview/x-road/Straumurinn–NotkunOgUmsysla.md` and you're good.

BTW. the file-naming should be in English. This PR contains two files with Icelandic names.

**Comment by bthj** [comment]
> Just need to format `handbook/technical-overview/x-road/Straumurinn–NotkunOgUmsysla.md` and you're good.
> 
> BTW. the file-naming should be in English. This PR contains two files with Icelandic names.

@darrikonn, what command would be suitable to format that particular file?

I've tried `nx format:check --all` and `nx format:check handbook/technical-overview/x-road/Straumurinn–NotkunOgUmsysla.md` without success.

I'll rename the files, but their content is in Icelandic, with links to the official X-Road documentation in English.

**Comment by darrikonn** [comment]
Got to skip the `:check` flag in order to format... otherwise you're linting.

**Comment by bthj** [comment]
> Got to skip the `:check` flag in order to format... otherwise you're linting.

Thanks @darrikonn, now we're green.

**Review by jeremybarbet — APPROVED** [review]

**Review by darrikonn — APPROVED** [review]

**Pull request merged**